core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -256,6 +272,15 @@ public boolean hasNext() {
     @Override
     public Location next() {
       int totalLength = PlatformDependent.UNSAFE.getInt(pageBaseObject, offsetInPage);
+      if (destructive) {
+        MemoryLocation keyAddress = loc.getKeyAddress();
+        Object keyBaseObject = keyAddress.getBaseObject();
+        long keyBaseOffset = keyAddress.getBaseOffset();
+
+        int hashcode = HASHER.hashUnsafeWords(keyBaseObject, keyBaseOffset, loc.getKeyLength());
+        int pos = hashcode & map.mask;
+        this.map.bitset.unset(pos);
+      }
```

Comment:
No, I just wanted to avoid the inconsistent state. As it is safe, I will remove this block.